docs/guide/widgets.md

```diff
@@ -388,3 +384,191 @@ The following builtin widgets use the Line API. If you are building advanced wid
+
+With these three widgets, the [DOM](CSS.md#the-dom) for our app will look like this:
+
+<div class="excalidraw">
+--8<-- "docs/images/byte_input_dom.excalidraw.svg"
+</div>
+
+Now that we have the design in place, we can implement the behavior.
+
+
+### Data flow
+
+We want to ensure that our widgets are re-usable, which we can do by following the guideline of "attributes down, messages up". This means that a widget can update a child by setting its attributes or calling its methods, but widgets should only ever send [messages](./events.md) to their *parent*.
+
+!!! info
+
+    This pattern of only setting attributes in one direction and using messages for the opposite direction is known as *uni-directional data flow*.
+
+In practice, this means that to update a child widget you get a reference to it and use it like any other Python object. Here's an example of an [action](actions.md) that updates a child widget:
+
+```python
+def action_set_true(self):
+    self.query_one(Switch).value = 1
+```
+
+If a child needs to update a parent, it should send a message with [post_message][textual.message_pump.MessagePump.post_message].
+
+Here's an example of posting message:
+
+```python
+def on_click(self):
+    self.post_message(MyWidget.Change(active=True))
+```
+
+Note that *attributes down and messages up* means that you can't modify widgets on the same level directly. If you want to modify a *sibling*, you will need to send a message to the parent, and the parent would make the changes.
+
+The following diagram illustrates this concept:
+
+
+<div class="excalidraw">
+--8<-- "docs/images/attributes_messages.excalidraw.svg"
+</div>
+
+### Messages up
+
+Let's extend the ByteEditor so that clicking any of the 8 `BitSwitch` widgets updates the decimal value. To do this we will add a custom message to `BitSwitch` that we catch in the `ByteEditor`.
+
+=== "byte02.py"
+
+    ```python title="byte02.py" hl_lines="5-6 26-32 34 44-48 91-96"
+    --8<-- "docs/examples/guide/compound/byte02.py"
+    ```
+
+    1. This will store the value of the "bit".
+    2. This is sent by the builtin `Switch` widgets, when it changes state.
+    3. Stop the event, because we don't want it to go to the parent.
+    4. Store the new value of the "bit".
+
+=== "Output"
+
+    ```{.textual path="docs/examples/guide/compound/byte02.py" columns="90" line="30", press="tab,tab,tab,tab,enter"}
+    ```
+
+- The `BitSwitch` widget now has an `on_switch_changed` which will handle a [Switch.Changed][textual.widgets.Switch.Changed] message, sent when the user clicks a switch. We use this to store the new value of the bit, and sent a new custom message, `BitSwitch.BitChanged`.
+- The `ByteEditor` widget handles the `BitSwitch.Changed` message by calculating the decimal value and setting it on the input.
+
+The following is a (simplified) DOM diagram to show how the new message is processed:
+
+<div class="excalidraw">
+--8<-- "docs/images/bit_switch_message.excalidraw.svg"
```

Comment:
Can the font size of the text be increased a bit? I'm worried it may be too small for some, especially given the brightness of the colours used.

Reply:
You're right, it is hard to read on mobile. I'll stack them on top of each other, that will help.